contrib/babelfishpg_common/src/datetimeoffset.c

```diff
@@ -639,6 +640,43 @@ timestamp_datetimeoffset(PG_FUNCTION_ARGS)
 	PG_RETURN_DATETIMEOFFSET(result);
 }
 
+/* timestamptz_datetimeoffset()
+ * Convert timestamp with time zone to datetimeoffset
+ */
+Datum
+timestamptz_datetimeoffset(PG_FUNCTION_ARGS)
```

Comment:
Suggestion: I would suggest to not define these function with `PG_FUNCTION_ARGS` until it is supposed to be invoked by SQL function. Invoking these functions from C using `DirectFunctionCall1` involves building `FCINFO` and will have downside of it in terms of perfomance.

Reply:
Done